File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/text/TextStyle.java
package top.likoslupus.cellulosesz.api.text;

public record TextStyle(
        String color,
        boolean bold,
        boolean italic,
        boolean underlined,
        boolean strikethrough,
        boolean obfuscated
) {

    public static final TextStyle EMPTY = new TextStyle(
            "",
            false,
            false,
            false,
            false,
            false
    );

    public TextStyle withColor(String color) {
        return new TextStyle(color, bold, italic, underlined, strikethrough, obfuscated);
    }

    public TextStyle withBold(boolean value) {
        return new TextStyle(color, value, italic, underlined, strikethrough, obfuscated);
    }

    public TextStyle withItalic(boolean value) {
        return new TextStyle(color, bold, value, underlined, strikethrough, obfuscated);
    }

    public TextStyle withUnderlined(boolean value) {
        return new TextStyle(color, bold, italic, value, strikethrough, obfuscated);
    }

    public TextStyle withStrikethrough(boolean value) {
        return new TextStyle(color, bold, italic, underlined, value, obfuscated);
    }

    public TextStyle withObfuscated(boolean value) {
        return new TextStyle(color, bold, italic, underlined, strikethrough, value);
    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/text/TextStyle.kt
package top.likoslupus.cellulosesz.api.text

@JvmRecord
public data class TextStyle(
    public val color: String,
    public val bold: Boolean,
    public val italic: Boolean,
    public val underlined: Boolean,
    public val strikethrough: Boolean,
    public val obfuscated: Boolean
) {

    public fun withColor(color: String): TextStyle = copy(color = color)

    public fun withBold(value: Boolean): TextStyle = copy(bold = value)

    public fun withItalic(value: Boolean): TextStyle = copy(italic = value)

    public fun withUnderlined(value: Boolean): TextStyle = copy(underlined = value)

    public fun withStrikethrough(value: Boolean): TextStyle = copy(strikethrough = value)

    public fun withObfuscated(value: Boolean): TextStyle = copy(obfuscated = value)

    public companion object {

        @JvmField
        public val EMPTY: TextStyle = TextStyle(
            color = "",
            bold = false,
            italic = false,
            underlined = false,
            strikethrough = false,
            obfuscated = false
        )

    }

}
